tests/foreman/longrun/test_inc_updates.py

```diff
@@ -0,0 +1,338 @@
+            reposet=PRD_SETS['rhel6_sat6tools']['reposet'],
+            releasever=PRD_SETS['rhel6_sat6tools']['releasever']
+        )
+
+        # Step 5: Read the repositories
+        cls.rhel_66_repo = Repository(id=rhel_66_repo_id).read()
+        cls.rhel6_sat6tools_repo = Repository(
+            id=rhel6_sat6tools_repo_id
+        ).read()
+
+        # Step 6: Sync the enabled repositories
+        try:
+            cls.old_task_timeout = entity_mixins.TASK_TIMEOUT
+            # Update timeout to 2 hours to finish sync
+            entity_mixins.TASK_TIMEOUT = 7200
+            for repo in [cls.rhel_66_repo, cls.rhel6_sat6tools_repo]:
+                response = Repository(id=repo.id).sync()
+                assert response['result'] == u'success'
+        finally:
+            entity_mixins.TASK_TIMEOUT = cls.old_task_timeout
+
+        # Step 7: Create two content views - one will be used with all erratas
+        # and another with erratas filtered
+        cv_temp = {}
+        for cv_name in [u'rhel_6_cv', u'rhel_6_partial_cv']:
+            cv_temp[cv_name] = ContentView(
+                organization=cls.org,
+                name=cv_name,
+                repository=[cls.rhel_66_repo, cls.rhel6_sat6tools_repo]
+            ).create()
+        cls.rhel_6_cv = cv_temp['rhel_6_cv']
+        cls.rhel_6_partial_cv = cv_temp['rhel_6_partial_cv']
+
+        # Step 8: Create a content view filter to filter out errata
+        rhel_6_partial_cvf = ErratumContentViewFilter(
+            content_view=cls.rhel_6_partial_cv,
+            type='erratum',
+            name='rhel_6_partial_cv_filter',
+            repository=[cls.rhel_66_repo]
+            ).create()
+
+        # Step 9: Create a content view filter rule - filtering out errata in
+        # the last 365 days
+        start_date = (date.today() - timedelta(days=365)).strftime('%Y-%m-%d')
+        ContentViewFilterRule(
+            content_view_filter=rhel_6_partial_cvf,
+            types=['security', 'enhancement', 'bugfix'],
+            start_date=start_date,
+            end_date=date.today().strftime('%Y-%m-%d')
+        ).create()
+
+        # Step 10: Publish both the content views and re-read the content views
+
+        # Changing the nailgun timeout time for the rest of the steps as
+        # publish/promote of larger repos take more than 5 minutes
+        entity_mixins.TASK_TIMEOUT = 3600
+        for content_view in [cls.rhel_6_cv, cls.rhel_6_partial_cv]:
+            content_view.publish()
+        cls.rhel_6_cv = cls.rhel_6_cv.read()
+        cls.rhel_6_partial_cv = cls.rhel_6_partial_cv.read()
+
+        # Step 10.a: Get the content view version ids of the published content
+        # views
+        cv_version_temp = {}
+        for content_view in [cls.rhel_6_cv, cls.rhel_6_partial_cv]:
+            cv_version_temp[content_view.name] = ContentView(
+                id=content_view.id
+            ).read_json()['versions']
```

Comment:
I think the 10.a and 10.b steps can be dramatically shortened. ContentView objects have a "version" attribute. You can use it like so:

```python
len(content_view.version)  # How many versions does this content view have?
content_view.version[0]  # Access a ContentViewVersion object.
```